hyper/common/headers.py

```diff
@@ -181,6 +181,28 @@ def iter_raw(self):
         for item in self._items:
             yield item
 
+    def replace(self, key, value):
+        """
+        Replace existing header with new value. If header doesn't exist this
+        method work like ``__setitem__``. Replacing leads to deletion of all 
+        exsiting headers with the same name.
+        """
+        idx = None
+        key = to_bytestring(key)
+        for (i, (k, v)) in enumerate(self._items):
+            if _keys_equal(k, key):
+                idx = i
+                break
+
+        if idx is not None:
+            del self[key]
```

Comment:
Given that we saved the index off, I think we can work on `_items` directly and just remove that index. That should save us from an extra loop over the container.